client/dashboard/me/tax-details/use-record-user-tax-events.ts

```diff
@@ -0,0 +1,41 @@
+import { useDispatch } from '@wordpress/data';
+import { useEffect, useRef } from 'react';
+import { useAnalytics } from '../../app/analytics';
+import type { UpdateError, FetchError } from './use-user-tax-details';
+
+export default function useRecordUserTaxEvents( {
+	updateError,
+	fetchError,
+	isUpdateSuccessful,
+}: {
+	updateError?: UpdateError | null;
+	fetchError?: FetchError | null;
+	isUpdateSuccessful?: boolean;
+} ) {
+	const reduxDispatch = useDispatch();
+	const lastFetchError = useRef< FetchError >();
+	const lastUpdateError = useRef< UpdateError >();
+	const { recordTracksEvent } = useAnalytics();
+
+	useEffect( () => {
```

Comment:
Recording tracks events inside `useEffect` callbacks is a bit implicit. You can see we need to do special things like keeping a ref to the old error just so we can check whether it has changed. These look like tracks events fired on success and failure, so we can place them inside the relevant mutation's `onSuccess` and `onError` callbacks.